development/verify-locale-strings.js

```diff
@@ -4,37 +4,87 @@
 //
 // usage:
 //
-//     node app/scripts/verify-locale-strings.js <locale>
+//     node app/scripts/verify-locale-strings.js [<locale>] [--fix]
 //
-// will check the given locale against the strings in english
+// This script will validate that locales have no unused messages. It will check
```

Comment:
It sounds like we might want the inverse also, based on the concerns you've raised: Validate that no UI template call uses a non-existent message. Not sure how efficiently this can be done, but if we assume the template string is always in a `t()` call, it might be efficient enough to regex.

Just nervous about removing used messages, like you suggested might be the case!

Reply:
I tried this already but too many cases use variables instead of string literals.

Many of those cases could be eliminated pretty easy, but there are a few that are very dynamic that would be challenging to refactor.

A better way to catch these cases might be to send an error to Sentry when we detect a missing key from the `en` locale, rather than just print a console warning like we do now.

Reply:
Yeah, that'd be a good use of sentry.